src/core/DrumSynth.cpp

```diff
@@ -258,7 +250,7 @@ int DrumSynth::GetPrivateProfileString(
 	if (len == 0)
 	{
 		len = strlen(def);
-		strncpy(buffer, def, size);
+		std::memcpy(buffer, def, std::min(len, size));
```

Comment:
Note that this changes the behavior. Before, the call to `strncpy` was copying `size` characters, but now its copying `std::min(len, size)` characters. I don't know why `len` was calculated here if it wasn't used before, but this does change the behavior slightly. Also, `strncpy` is bounded already, so maybe we do not need to make such changes.

Reply:
This is a behavioral change, because `buffer`, `size`, and `def` are all function parameters. This change prevents problems when `strlen(def) > size`, which is possible.

`len` is a member variable, which is why it was calculated there before.

`strncpy()` is bounded in the sense it won't write past its bounds, but if the bounds are reached it will not null-terminate the copied string. `strncpy()` is not a safer `strcpy()` without actually making sure the string is null terminated.

Looking at the callsites of this function makes it apparent it should have been `std::min(len, size - 1)`, otherwise this has the same problem lol

Reply:
The DrumSynth code is a dark labyrinth in and of itself, so I naturally am less inclined to change it. But if you are confident the logic is sound, its fine with me.